v0/destinations/google_adwords_remarketing_lists/transform.js

```diff
@@ -132,66 +132,59 @@ const populateIdentifiers = (attributeArray, { Config }) => {
 
 const createPayload = (message, destination) => {
   const { listData } = message.properties;
+  const properties = ["add", "remove"];
 
-  if (!listData.add || !isArray(listData.add)) {
-    throw new CustomError(
-      "[Google_adwords_remarketing_list]::add is not present inside listData. Aborting message.",
-      400
-    );
-  }
-  const userIdentidtifiersList = populateIdentifiers(listData.add, destination);
-  const outputPayload = constructPayload(message, offlineDataJobsMapping);
-  outputPayload.operations = [];
-  // breaking the userIdentiFier array in chunks of 20
-  const userIdentifierChunks = returnArrayOfSubarrays(
-    userIdentidtifiersList,
-    20
-  );
-  // putting each chunk in different create/remove operations
-  userIdentifierChunks.forEach(element => {
-    const operations = {
-      create: {}
-    };
-    operations.create.userIdentifiers = element;
-    outputPayload.operations.push(operations);
+  let outputPayloads = {};
+  const typeOfOperation = Object.keys(listData);
+  typeOfOperation.forEach(key => {
+    if (properties.includes(key)) {
+      const userIdentifiersList = populateIdentifiers(
+        listData[key],
+        destination
+      );
+      const outputPayload = constructPayload(message, offlineDataJobsMapping);
+      outputPayload.operations = [];
+      // breaking the userIdentiFier array in chunks of 20
+      const userIdentifierChunks = returnArrayOfSubarrays(
+        userIdentifiersList,
+        20
+      );
+      // putting each chunk in different create/remove operations
+      switch (key) {
+        case "add":
+          // for add operation
+          userIdentifierChunks.forEach(element => {
+            const operations = {
+              create: {}
+            };
+            operations.create.userIdentifiers = element;
+            outputPayload.operations.push(operations);
+          });
+          outputPayloads = { ...outputPayloads, create: outputPayload };
+          break;
+        case "remove":
+          // for remove operation
+          userIdentifierChunks.forEach(element => {
+            const operations = {
+              remove: {}
+            };
+            operations.remove.userIdentifiers = element;
+            outputPayload.operations.push(operations);
+          });
+          outputPayloads = { ...outputPayloads, remove: outputPayload };
+          break;
+        default:
+      }
+    } else {
+      logger.log(`listData's ${key} is not valid.`);
```

Comment:
Can you update this message with something like, "we are supporting add and remove only ${key} is not supported."

Reply:
Modified